torch/_dynamo/variables/torch.py

```diff
@@ -757,10 +764,27 @@ def safe_or_raise_always_restore(tx, graph_checkpoint, checkpoint, f, sub_args):
         tx.restore_graphstate(checkpoint)
 
 
+@contextlib.contextmanager
+def dynamo_enable_grad(tx):
+    from . import GradModeVariable
+
+    org_value = torch.is_grad_enabled()
+    GradModeVariable.create(tx, True)
+    yield
+    GradModeVariable.create(tx, org_value)
```

Comment:
Does this need:
```
org_value = torch.is_grad_enabled()
try:
  GradModeVariable.create(tx, True)
  yield
finally:
   GradModeVariable.create(tx, org_value)
```
?
(that's what other context managers using contextlib.contextmanager do, but I'm not sure if we need anything special in this case)

Reply:
Given that we bail out and restore back from checkpoint if there are errors while tracing the function, I thought this would be ok. (But agreed that using with try..finally seems more inline with the codebase, will update this)